frontend/ui/widget/inputtext.lua

```diff
@@ -596,43 +609,35 @@ function InputText:onKeyPress(key)
     local handled = true
 
     if not key["Ctrl"] and not key["Shift"] and not key["Alt"] and not key["ScreenKB"] then
-        if key["Backspace"] then
-            self:delChar()
+        if key["Backspace"] then self:delChar()
         elseif key["Del"] then
             -- Kindles with physical keyboards only have a "Del" key (no "Backspace").
             if Device:hasSymKey() then
                 self:delChar()
             else
                 self:delNextChar()
             end
-        elseif key["Left"] then
-            self:leftChar()
-        elseif key["Right"] then
-            self:rightChar()
-        elseif key["Up"] then
+        elseif key["Left"] then self:leftChar()
+        elseif key["Right"] then self:rightChar()
+        -- NOTE: When we are not showing the virtual keyboard, up/down keys are used to directly move around the widget,
+        --       in and out of text fields and onto virtual buttons like `[cancel] [search dict]`, no need to unfocus first.
+        elseif key["Up"] and G_reader_settings:nilOrTrue("virtual_keyboard_enabled") then
             self:upLine()
-        elseif key["Down"] then
+        elseif key["Down"] and G_reader_settings:nilOrTrue("virtual_keyboard_enabled") then
             self:downLine()
-        elseif key["End"] then
-            self:goToEnd()
-        elseif key["Home"] then
-            self:goToHome()
-        elseif key["Press"] then
-            self:addChars("\n")
-        elseif key["Tab"] then
-            self:addChars("    ")
-        elseif key["Back"] then
-            if self.focused then
-                self:unfocus()
-            end
+        elseif key["End"] then self:goToEnd()
+        elseif key["Home"] then self:goToHome()
+        elseif key["Press"] then self:addChars("\n")
```

Comment:
That's not obviously prettier, and touching lines that you don't need to. It was already hard to figure out what this PR is doing :/

Reply:
Will change them back then, pretty is in the eye of the beholder. But that function is a padding.tiny bit messy ba dam dam tsh.